Cache/IO.lean

```diff
@@ -9,31 +9,32 @@ import Lean.Data.RBMap
 import Lean.Data.RBTree
 import Lean.Data.Json.Printer
 import Lean.Data.Json.Parser
+import Cache.Lean
 
 variable {α : Type}
 
-/-- Removes a parent path from the beginning of a path -/
-def System.FilePath.withoutParent (path parent : FilePath) : FilePath :=
-  let rec aux : List String → List String → List String
-    | z@(x :: xs), y :: ys => if x == y then aux xs ys else z
-    | [], _ => []
-    | x, [] => x
-  mkFilePath <| aux path.components parent.components
-
-def Nat.toHexDigits (n : Nat) : Nat → (res : String := "") → String
-  | 0, s => s
-  | len+1, s =>
-    let b := UInt8.ofNat (n >>> (len * 8))
-    Nat.toHexDigits n len <|
-      s.push (Nat.digitChar (b >>> 4).toNat) |>.push (Nat.digitChar (b &&& 15).toNat)
-
-def UInt64.asLTar (n : UInt64) : String :=
-  s!"{Nat.toHexDigits n.toNat 8}.ltar"
+open Lean
 
 namespace Cache.IO
 
 open System (FilePath)
 
+/--
+Read the search path from `LEAN_PATH` and drop the trailing `/.lake/build/lib/`
+as the `.lean`-files are located outside the `.lake/` folders.
+-/
+def getCleanSearchPath : IO SearchPath := do
+  let sp ← addSearchPathFromEnv {}
+  return sp.map (System.mkFilePath <| dropLastThree ·.components)
+where
+  /-- helper function to remove last 3 entries, namely `.lake`, `build`, and `lib`. -/
+  dropLastThree  : List String → List String
+    | []    => []
+    | [_]   => []
+    | [_, _]   => []
+    | [_, _, _]   => []
+    | a :: rest@(_ :: _ :: _) => a :: dropLastThree rest
```

Comment:
Feels a bit overkill. Can you just `take l (l.length - 3)`?

I guess we're going to need `dropRight` and `takeRight` eventually.

Reply:
```
return sp.map fun path =>
    System.mkFilePath (path.components |> fun p => p.take (p.length - 3))
```
it does loop over each path's components twice this way, but honestly these aren't super long.